### PR TITLE
[CSV-265] Update buffer position when reading line comment

### PR DESCRIPTION
**Pull request by king-tyler**
https://issues.apache.org/jira/browse/CSV-265

### CODE REVIEW

**Comment by garydgregory** [comment]
@king-tyler 
May you please rebase on master?

**Comment by king-tyler** [comment]
@garydgregory 
I rebased on master

File: src/test/java/org/apache/commons/csv/issues/JiraCsv265Test.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    @Test
+    public void testJiraCsv265Test() throws IOException {
+        String csv = "# Comment1\n"
+                   + "Header1,Header2\n"
+                   + "# Comment2\n"
+                   + "Value1,Value2\n"
+                   + "# Comment3\n"
+                   + "Value3,Value4\n"
+                   + "# Comment4\n";
+        CSVFormat csvFormat = CSVFormat.DEFAULT.withCommentMarker('#').withFirstRecordAsHeader();
```
Comment (garydgregory) [inline src/test/java/org/apache/commons/csv/issues/JiraCsv265Test.java:42]: @king-tyler 
May you please use the builder instead of the deprecated methods?

Comment (king-tyler) [reply src/test/java/org/apache/commons/csv/issues/JiraCsv265Test.java]: Fixed

File: src/test/java/org/apache/commons/csv/issues/JiraCsv265Test.java [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+            .setSkipHeaderRecord(true)
+            .build();
+        // @formatter:on
+        try (final CSVParser parser = csvFormat.parse(new StringReader(csv))) {
+            final Iterator<CSVRecord> itr = parser.iterator();
+            final CSVRecord record1 = itr.next();
+            assertEquals(csv.indexOf("# Comment2"), record1.getCharacterPosition());
+            final CSVRecord record2 = itr.next();
+            assertEquals(csv.indexOf("# Comment3"), record2.getCharacterPosition());
+        }
```
Comment (garydgregory) [inline src/test/java/org/apache/commons/csv/issues/JiraCsv265Test.java:58]: I'm trying to better understand what you are trying to do here... may you please add a test with comments that span multiple lines; for example, input like:
```
# one
# two
data,data,data
# three
# foor
data,data,data
```

TY.

Comment (king-tyler) [reply src/test/java/org/apache/commons/csv/issues/JiraCsv265Test.java:58]: Fixed

**Pull request merged**